util/container/image.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package container
+
+import (
+	"fmt"
+	"path"
+	"regexp"
+
+	//  Import the crypto sha256 algorithm for the docker image parser to work
+	_ "crypto/sha256"
+	//  Import the crypto/sha512 algorithm for the docker image parser to work with 384 and 512 sha hashes
+	_ "crypto/sha512"
+
+	"github.com/docker/distribution/reference"
+	"github.com/pkg/errors"
+)
+
+var (
+	ociTagAllowedChars = regexp.MustCompile(`[^-a-zA-Z0-9_\.]`)
+)
+
+// Image type represents the container image details
+type Image struct {
+	Repository string
+	Name       string
+	Tag        string
+	Digest     string
+}
+
+// ImageFromString parses a docker image string into three parts: repo, tag and digest.
+// If both tag and digest are empty, a default image tag will be returned.
+func ImageFromString(image string) (Image, error) {
+	named, err := reference.ParseNamed(image)
```

Comment:
In the previous implementation, we used `reference.ParseNormalizedNamed(image)`. Is there some essential difference between `ParseNormalizedNamed` and `ParseName`?

Reply:
ParseName does more validation

Reply:
Good catch. `ParseNormalizedNamed` will allow for "Docker hub"-style names, ala `busybox:latest`. `ParseNamed` will error `ErrNameNotCanonical` on those. I don't know if we want to support images like that.

Reply:
I found a few cases with checks to make sure  we don't get ErrNameNotCanonical so I assumed its better to use ParsedName

Reply:
I think it's better just that it forces people to be explicit, the Docker default registry nonsense is terrible

Reply:
agreed, I took the default out